index.coffee

```diff
@@ -69,18 +12,25 @@ module.exports =
       default: ''
 
   activate: ->
-    prefix = 'linter-rubocop.'
     @subscriptions = new CompositeDisposable
-    @subscriptions.add atom.config.observe "#{prefix}executablePath",
-      (args) => @executablePath = if args then args.split ' ' else ['rubocop']
-    @subscriptions.add atom.config.observe "#{prefix}additionalArguments",
-      (args) => @additionalArguments = if args then args.split ' ' else []
+    @subscriptions.add atom.config.observe('linter-rubocop.executablePath', (executablePath) =>
+      @executablePath = executablePath
+    )
+    @subscriptions.add atom.config.observe('linter-rubocop.additionalArguments', (extraArguments) =>
+      @arguments = extraArguments.split(' ').filter((i) => i)
+    )
 
   deactivate: ->
     @subscriptions.dispose()
 
   provideLinter: ->
-    grammarScopes: ['source.ruby', 'source.ruby.rails', 'source.ruby.rspec', 'source.ruby.chef'],
-    scope: 'file'
-    lintOnFly: true
-    lint: (editor) => lint editor, @executablePath, @additionalArguments
+    helpers = require('atom-linter')
+    provider =
+      grammarScopes: ['source.ruby', 'source.ruby.rails', 'source.ruby.rspec', 'source.ruby.chef']
+      scope: 'file'
+      lintOnFly: true
+      lint: (textEditor) =>
+        return helpers.exec(@executablePath, ['-f', 'json', '-s', textEditor.getPath()].concat(@arguments),
+          {stdin: textEditor.getText()}).then (contents) ->
```

Comment:
We need to set the `cwd` to the nearest parent directory of the file somewhere in here.